crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -1154,6 +1157,80 @@ fn handle_attribute_comment<'a>(
     }
 }
 
+/// Assign comments between `if` and `test` and `else` and `orelse` as leading to the respective
+/// node.
+///
+/// ```python
+/// x = (
+///     "a"
+///     if # leading comment of `True`
+///     True
+///     else # leading comment of `"b"`
+///     "b"
+/// )
+/// ```
+///
+/// This placement ensures comments remain in their previous order. This an edge case that only
+/// happens if the comments are in a weird position but it also doesn't hurt handling it.
+fn handle_expr_if_comment<'a>(
+    comment: DecoratedComment<'a>,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    let Some(expr_if) = comment.enclosing_node().expr_if_exp() else {
+        return CommentPlacement::Default(comment);
+    };
+    let ExprIfExp {
+        range: _,
+        test,
+        body,
+        orelse,
+    } = expr_if;
+
+    // Find the if and the else
+    let if_token =
+        find_only_token_str_in_range(TextRange::new(body.end(), test.start()), locator, "if");
+    let else_token =
+        find_only_token_str_in_range(TextRange::new(test.end(), orelse.start()), locator, "else");
+
+    if if_token.range.start() < comment.slice().start()
+        && comment.slice().start() < test.start()
+        && comment.line_position().is_end_of_line()
```

Comment:
I recommend moving the `line_position` check to the very top to avoid unnecessarily lexing own line comments.

Reply:
good point